OptimizelySDK/Optimizely.cs

```diff
@@ -1093,13 +1352,54 @@ public OptimizelyConfig GetOptimizelyConfig()
                 return ((IOptimizelyConfigManager)ProjectConfigManager).GetOptimizelyConfig();
             }
 
-            Logger.Log(LogLevel.DEBUG, "ProjectConfigManager is not instance of IOptimizelyConfigManager, generating new OptimizelyConfigObject as a fallback");
+            Logger.Log(LogLevel.DEBUG,
+                "ProjectConfigManager is not instance of IOptimizelyConfigManager, generating new OptimizelyConfigObject as a fallback");
 
             return new OptimizelyConfigService(config).GetOptimizelyConfig();
         }
 
         #endregion FeatureFlag APIs
 
+        /// <summary>
+        /// Attempts to fetch and return a list of a user's qualified segments.
+        /// </summary>
+        /// <param name="userId">FS User ID</param>
+        /// <param name="segmentOptions">Options used during segment cache handling</param>
+        /// <returns>Qualified segments for the user from the cache or the ODP server</returns>
+        public string[] FetchQualifiedSegments(string userId, List<OdpSegmentOption> segmentOptions)
+        {
+            return OdpManager.FetchQualifiedSegments(userId, segmentOptions);
+        }
+
+        /// <summary>
+        /// Send identification event to Optimizely Data Platform
+        /// </summary>
+        /// <param name="userId">FS User ID to send</param>
+        public void IdentifyUser(string userId)
+        {
+            OdpManager.IdentifyUser(userId);
+        }
+
+        /// <summary>
+        /// Add event to queue for sending to ODP
+        /// </summary>
+        /// <param name="type">Type of event (typically `fullstack` from server-side SDK events)</param>
+        /// <param name="action">Subcategory of the event type</param>
+        /// <param name="identifiers">Key-value map of user identifiers</param>
+        /// <param name="data">Event data in a key-value pair format</param>
+        public void SendOdpEvent(string type, string action, Dictionary<string, string> identifiers,
+            Dictionary<string, object> data
+        )
+        {
+            if (OdpManager == null)
+            {
+                Logger.Log(LogLevel.ERROR, Constants.ODP_NOT_ENABLED_MESSAGE);
+                return;
+            }
```

Comment:
Why don't we have this guard for "fetchQualifiedSegments" and "identifyUser" above?

Reply:
Oops, I meant to remove this guard here in `SendOdpEvent` too. Instantiating `OdpManager` should be guaranteed via `InitializeComponents` call be constructor.

Should add the guards back, just in case things change?  What's your thought?

Reply:
No need for guard here if it's guaranteed non-null.